src/lidar_tools/dsm_functions.py

```diff
@@ -5,23 +5,31 @@
 from rasterio.warp import transform_bounds
 from pyproj import CRS
 import shapely
+from shapely.geometry import Polygon
 import geopandas as gpd
 import requests
 import rasterio
 import xarray as xr
 import rioxarray
 import pystac_client
 import numpy as np
-
+import json
+from pathlib import Path
+import glob
 # import planetary_computer
 from osgeo import gdal, gdalconst
+import pdal
 import odc.stac
 import os
+import gc
+
 
 gdal.UseExceptions()
 
 odc.stac.configure_rio(cloud_defaults=True)
 
+release_second =  10 # seconds to wait before releasing the lock in parallel processing
+
```

Comment:
:) Ideally we'll figure out an approach that doesn't require manual fiddling with locks